src/pkg/token/claims/v2/claims.go

```diff
@@ -39,7 +40,9 @@ type Claims struct {
 
 // Valid checks if the issuer is harbor
 func (c *Claims) Valid() error {
-	if err := c.RegisteredClaims.Valid(); err != nil {
+	var v = jwt.NewValidator(jwt.WithLeeway(60 * time.Second))
+
+	if err := v.Validate(c.RegisteredClaims); err != nil {
 		return err
 	}
 	if subtle.ConstantTimeCompare([]byte(c.Issuer), []byte(Issuer)) == 0 {
```

Comment:
we could probably remove this `subtle.ConstantTimeCompare` part with  `jwt.NewValidator(jwt.WithIssuer(c.Issuer))`. As it all aims to compare the issuer.

Reply:
Should be addressed by 899634c